src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java

```diff
@@ -0,0 +1,20 @@
+package org.folio.de.entity.bursarlegacy;
+
+import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.folio.de.entity.BaseJob;
+import org.folio.des.domain.dto.LegacyExportTypeSpecificParameters;
+import org.hibernate.annotations.Type;
+
+@Entity
+@Table(name = "job")
+@Data
+public class LegacyJob extends BaseJob {
```

Comment:
For example we use this table for edifact export, and that jobs for us is not legacy jobs, and our parameters is not legacy.
I think we can simplify this. You can remain 1 class without this refactoring but introduce custom jsonb converter that can convert to generic jsonb to both dtos.

Here is example of usage:
`ublic class PersonCompanyJsonbConverter implements AttributeConverter<Object, Jsonb> {
 
    private static final Jsonb JSONB = JsonbBuilder.create();
 
    @Override
    public Jsonb convertToDatabaseColumn(Object object) {
        return JSONB.toJson(object);
    }
 
    @Override
    public Object convertToEntityAttribute(Jsonb jsonb) {
        JsonArray jsonArray = jsonb.fromJson(JsonArray.class);
        if (jsonArray.size() > 0) {
            JsonObject jsonObject = jsonArray.getJsonObject(0);
            if (jsonObject.containsKey("name")) {
                return JSONB.fromJson(jsonb.toString(), Person.class);
            } else if (jsonObject.containsKey("address")) {
                return JSONB.fromJson(jsonb.toString(), Company.class);
            }
        }
        throw new IllegalArgumentException("Cannot deserialize JSONB to Person or Company class.");
    }
}`


and how to use:
`@Entity
@Table(name = "my_data")
public class MyData {
 
    @Id
    private Long id;
 
    @Column
    @Convert(converter = PersonCompanyJsonbConverter.class)
    private Object data;
 
    // getters and setters
}`

Reply:
So if I understand it correctly, you want me to keep just one `Job` class instead of splitting it into `LegacyJob` and `Job` and use  a custom jsonb converter that can convert to generic jsonb to both dtos, right? I don't think I fully understand how you want me to change this code. Would we then have two different `ExportTypeSpecificParameters` attributes in the `Job` class?

Reply:
we can use some abstart class for like BaseExportTypeSpecificParameters and in converter chose what instance to create:
ExportTypeSpecificParameters or LegacyExportTypeSpecificParameters based of content of jsonb field.
But if it hard to implement I am ok with this approach with mapped superclass.

Reply:
@SerhiiNosko I have tried implementing it as above like you said but I ran into problems with generating correct Java classes using OpenAPI generator for the abstract class and inherited classes like above. I have described it more in details in DM if you can take a look.